src/Appwrite/Platform/Tasks/Restore.php

```diff
@@ -0,0 +1,291 @@
+<?php
+
+namespace Appwrite\Platform\Tasks;
+
+use Exception;
+use Utopia\App;
+use Utopia\DSN\DSN;
+use Utopia\Platform\Action;
+use Utopia\CLI\Console;
+use Utopia\Storage\Device;
+use Utopia\Storage\Device\DOSpaces;
+use Utopia\Storage\Device\Local;
+use Utopia\Validator\Text;
+use Utopia\Validator\WhiteList;
+
+class Restore extends Action
+{
+    protected ?DSN $dsn = null;
+    protected string $database;
+    public const PROCESSORS = 4;
+
+    public static function getName(): string
+    {
+        return 'restore';
+    }
+
+    public function __construct()
+    {
+        $this->checkEnvVariables();
+
+        $this
+            ->desc('Restore a DB')
+            ->param('id', '', new Text(20), 'Folder Identifier')
+            ->param('cloud', null, new WhiteList(['true', 'false'], true), 'Take file from cloud?')
+            ->param('database', null, new Text(10), 'example: db_fra1_01..')
+            ->param('datadir', null, new Text(100), 'mysql datadir path')
+            ->callback(fn ($id, $cloud, $project, $datadir) => $this->action($id, $cloud, $project, $datadir));
+    }
+
+    public function action(string $id, string $cloud, string $database, string $datadir): void
+    {
+        $this->database = $database;
+        $this->dsn = Backup::getDsn($database);
+
+        if (!$this->dsn instanceof DSN) {
+            Console::error('No dsn match');
+            Console::exit();
+        }
+
+        if (!file_exists($datadir)) {
+            Console::error('Datadir not found: ' . $datadir);
+            Console::exit();
+        }
+
+        if (file_exists($datadir . '/sys') || file_exists($datadir . '/appwrite')) {
+            Console::error('Datadir ' . $datadir . ' must be empty!');
+            Console::exit();
+        }
+
+        //todo: check if the container is stopped
+        //$this->checkContainerIsStopped();
```

Comment:
Do we need this check or can we remove this?

Reply:
Done!